packages/node/src/subcommands/testing.service.ts

```diff
@@ -0,0 +1,253 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { assert } from 'console';
+import { readdirSync, statSync } from 'fs';
+import path from 'path';
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  NodeConfig,
+  StoreService,
+  getLogger,
+  SandboxOption,
+  TestSandbox,
+  IndexerSandbox,
+} from '@subql/node-core';
+import { SubqlTest } from '@subql/testing/interfaces';
+import {
+  DynamicDatasourceCreator,
+  Entity,
+  Store,
+  SubstrateHandlerKind,
+} from '@subql/types';
+import { getAllEntitiesRelations } from '@subql/utils';
+import Pino from 'pino';
+import { CreationAttributes, Model, Options, Sequelize } from 'sequelize';
+import { register, RegisterOptions } from 'ts-node';
+import { SubqueryProject } from '../configure/SubqueryProject';
+import { ApiService } from '../indexer/api.service';
+import { IndexerManager } from '../indexer/indexer.manager';
+import * as SubstrateUtil from '../utils/substrate';
+
+const logger = getLogger('CLI-test-Testing');
+
+declare global {
+  //const api: ApiAt;
+  const logger: Pino.Logger;
+  const store: Store;
+  const createDynamicDatasource: DynamicDatasourceCreator;
+}
+
+@Injectable()
+export class TestingService {
+  private tests: SubqlTest[];
+  constructor(
+    private readonly sequelize: Sequelize,
+    private readonly nodeConfig: NodeConfig,
+    private readonly storeService: StoreService,
+    @Inject('ISubqueryProject') private project: SubqueryProject,
+    private readonly apiService: ApiService,
+    private readonly indexerManager: IndexerManager,
+  ) {}
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async init() {
+    const projectPath = this.project.root;
+    // find all paths to test files
+    const testFiles = this.findAllTestFiles(
+      path.join(projectPath, 'dist/test'),
+    );
+    // import all test files
+
+    const { compilerOptions } = require(path.join(
+      projectPath,
+      'tsconfig.json',
+    ));
+
+    const options: RegisterOptions = {
+      compilerOptions,
+      files: true,
+    };
+
+    register(options);
```

Comment:
Im not sure why we're doing anything with TS

Reply:
we don't need these anymore, deleting